core/src/main/java/com/netflix/conductor/core/execution/ParametersUtils.java

```diff
@@ -223,9 +223,9 @@ private String getSystemParametersValue(String sysParam, String taskId){
 		if("CPEWF_TASK_ID".equals(sysParam)) {
 			return taskId;
 		}
-		String value = System.getProperty(sysParam);
+		String value = System.getenv(sysParam);
 		if(value == null) {
-			value = System.getenv(sysParam);
+			value = System.getProperty(sysParam);
 		}
```

Comment:
This is enforcing the order of precedence as 1. environment variable and then 2. configuration file. This establishes the configuration file as defaults and environment variables as overrides.